app/src/main/java/org/fossasia/openevent/app/common/di/module/BarcodeModule.java

```diff
@@ -0,0 +1,69 @@
+package org.fossasia.openevent.app.common.di.module;
+
+import android.content.Context;
+import android.util.SparseArray;
+
+import com.google.android.gms.vision.CameraSource;
+import com.google.android.gms.vision.Detector;
+import com.google.android.gms.vision.barcode.Barcode;
+import com.google.android.gms.vision.barcode.BarcodeDetector;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import io.reactivex.Notification;
+import io.reactivex.subjects.PublishSubject;
+
+@Module(includes = AndroidModule.class)
+public class BarcodeModule {
+
+    @Provides
+    @Singleton
+    @Named("barcodeEmitter")
+    PublishSubject<Notification<Barcode>> providesBarcodeEmitter() {
+        return PublishSubject.create();
+    }
+
+    @Provides
```

Comment:
Creating singleton is not a good practice in Java, but we need it for modules which will be used in several views throughout the app. Cameras and presenters are not such. They are mapped 1:1 to the views and also, they are bound to a very tight lifecycle. Thus, can't reside in memory forever